Mathlib/ModelTheory/Semantics.lean

```diff
@@ -1023,6 +1042,301 @@ theorem infinite [Mi : Infinite M] (h : M ≅[L] N) : Infinite N :=
 
 end ElementarilyEquivalent
 
+section Definability
+
+variable [inst' : L'.Structure M]
+
+namespace Term
+
+/-- `Term.subst_definitions` agrees with the original formula once realized, assuming all the side
+conditions are met. -/
+theorem subst_definitions_eq (t : L.Term α)
+  {Fs : ∀ {n} (_ : L.Functions n), L'.Formula (Fin n ⊕ Unit)}
+  (hFs : ∀ {n} (g : L.Functions n),
+    (Function.tupleGraph fun v ↦ g.term.realize v) = ((Fs g).Realize : Set (_ → M)))
+  {sideVals : Fin (t.subst_definitions Fs).1 → M}
+  (v : α → M)
+  (hSideVals : ∀ s ∈ (t.subst_definitions Fs).2.2, s.Realize (Sum.elim v sideVals))
+  : (t.subst_definitions Fs).2.1.realize (Sum.elim v sideVals) = t.realize v := by
+    induction t
+    next a =>
```

Comment:
This is not the recommended style afaik.

Reply:
What is the preferred style? I see a lot of different styles in mathlib when I do a quick search. (`induction` followed by `·`, followed by `| zero =>`, even a few instances of `· next x y =>`...)